azure-pipelines/pull-request-validation/build-consumers.yml

```diff
@@ -0,0 +1,193 @@
+# File: azure-pipelines\pull-request-validation\build-consumers.yml
+# Description: Test new common in ADAL / MSAL / Broker (assembleLocal and testLocalDebugUnitTest)
+# Variable: 'ENV_VSTS_MVN_ANDROIDADACCOUNTS_USERNAME' was defined in the Variables tab
+# Variable: 'mvnAccessToken' was defined in the Variables tab
+# https://docs.microsoft.com/en-us/azure/devops/organizations/accounts/use-personal-access-tokens-to-authenticate
+# Variable: 'commonBranchName' common branch to be used when running the pipeline manually
+name: $(date:yyyyMMdd)$(rev:.r)
+
+trigger: none
+
+variables:
+- name: robolectricSdkVersion
+  value: 28
+
+resources:
+  repositories:
+  - repository: msal
+    type: github
+    name: AzureAD/microsoft-authentication-library-for-android
+    ref: dev
+    endpoint: ANDROID_GITHUB
+  - repository: broker
+    type: github
+    name: AzureAD/ad-accounts-for-android
+    ref: dev
+    endpoint: ANDROID_GITHUB
+  - repository: adal
+    type: github
+    name: AzureAD/azure-activedirectory-library-for-android
+    ref: dev
+    endpoint: ANDROID_GITHUB
+
+stages:
+- stage: validateConsumers
+  displayName: Validate Consumer
+  pool:
+    name: Hosted Windows 2019 with VS2019
+  jobs:
+  # setup
+  - job: setupBranch
+    displayName: Setup branch
+    steps:
+    - checkout: none
+    - task: PowerShell@2
+      name: setvarStep
+      displayName: Set var branch
+      inputs:
+        targetType: inline
+        script: |
+          $branch = switch ( '$(Build.Reason)' )
+          {
+              PullRequest
+              {
+                '$(System.PullRequest.SourceBranch)'
+              }
+              Manual
+              {
+                '$(commonBranchName)'
+              }
+              default
+              {
+                ''
+              }
+          }
+          Write-Output "$branch"
+          echo "##vso[task.setvariable variable=commonBranch;isOutput=true]$branch"
+    - powershell: |
+        if ("$(setvarStep.commonBranch)" -eq "") {
+          Write-Output "commonBranchName is not set using default common"
+        }
+        else{
+          Write-Output "$(setvarStep.commonBranch)"
+        }
+      name: echovar
+      displayName: Echo branch name
+  # msal
+  - job: msalValidation
+    displayName: MSAL
+    dependsOn: setupBranch
+    variables:
+      commonBranch: $[ dependencies.setupBranch.outputs['setvarStep.commonBranch'] ]  # map in the variable
+    steps:
+    - template: ../templates/steps/automation-cert.yml
+    - checkout: msal
+      displayName: Checkout msal repository
+      clean: true
+      submodules: recursive
+      persistCredentials: True
+    - task: CmdLine@2
+      displayName: Checkout common submodule $(commonBranch)
+      inputs:
+        script: |
+          git fetch
+          git checkout $(commonBranch)
```

Comment:
Do we need to also do a `git pull`? Or since this is a fresh checkout it should bring the latest commit anyway?

Reply:
it is a fresh checkout so it is the first time is pulling the branch and I am fetching before the checkout, I don't think is really necessary but I also don't think it affects to add it, I will do it